src/Modules/Station/Components/OnlineUsers/OnlineUsers.tsx

```diff
@@ -137,43 +172,51 @@ export class OnlineUsersComponent extends BaseComponent<IProps, IState> {
         onClick={this.toggle}>
         <i
           className={classNames('fa', {
-            'fa-circle online-color': fixture.length > 0,
-            'fa-circle-o': fixture.length <= 0,
+            'fa-circle online-color': numberOnline > 0,
+            'fa-circle-o': numberOnline <= 0,
           })}
         />
         <span className="online-users-length">
-          {fixture.length || '0'} online
+          {numberOnline || '0'} online
         </span>
       </div>,
-      <div key={3}>{this.renderOnlineTooltip(fixture, 'online-users')}</div>,
+      <div key={3}>
+        {this.renderOnlineTooltip(
+          filteredListWithoutAnonymous,
+          numberOnline,
+          'online-users',
+        )}
+      </div>,
       <Popover
         key={2}
         placement="bottom"
         isOpen={this.state.popoverOpen}
         toggle={this.toggle}
         target="online-users">
-        {this.renderPopoverContent()}
+        {this.renderPopoverContent(filteredListWithoutAnonymous, numberOnline)}
       </Popover>,
     ];
   }
 
-  private renderOnlineTooltip(list: any[], target: string) {
-    if (this.state.popoverOpen) return null;
-    return (
-      <UncontrolledTooltip placement="bottom" target={target}>
-        {list.map(({ name, username }, index) => (
-          <div className="online-tooltip" key={index}>
-            <span>
-              {this.isUserInfoAvailable(currentUser) && this.isMe(username)
-                ? 'You'
-                : name}
-            </span>
-            <br />
-          </div>
-        ))}
-      </UncontrolledTooltip>
-    );
-  }
+  private toArray = (userMap: any) => {
```

Comment:
use Type of onlineUsers instead of any after declaring.
Please check for removeAnonymousFromArray function